docs/content/posts/2024-10-01-release-2.0-preview.md

```diff
@@ -0,0 +1,1264 @@
+- **TableSchema, TableColumn and Types.** Please migrate to [Schema](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/Schema.java), [Column](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/Column.java) and [DataTypes](https://github.com/apache/flink/blob/master/flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/DataTypes.java) respectively.
+
+Some deprecated methods have been removed from **DataStream API**. See also the list of [removed programming APIs](#list-of-removed-programming-apis).
+
+Some deprecated fields have been removed from **REST API**. See also the list of [removed REST APIs](#list-of-removed-rest-apis).
+
+**NOTICE:** You may find some of the removed APIs still exist in the code base, usually in a different package. They are for internal usages only and can be changed / removed anytime without notifications. Please **DO NOT USE** them.
+
+### Connector Adaption Plan
+
+As SourceFunction, SinkFunction and SinkV1 being removed, existing connectors depending on these APIs will not work on the Flink 2.x series. Here's the plan for adapting the first-party connectors. 
+1. A new version of Kafka connector, adapted to the API changes, will be released right after the release of Flink 2.0 Preview.
+2. JDBC and ElasticSearch connectors will be adapted by the formal release of Flink 2.0.
+3. We plan to gradually migrate the remaining first-party connectors within 3 subsequent minor releases (i.e., by Flink 2.3).
+
+## Configuration
+
+Configuration options meet the following criteria are removed. See also the list of [removed configuration options](#list-of-removed-configuration-options).
+- Annotated as `@Public` and have been deprecated for at least 2 minor releases.
+- Annotated as `@PublicEvolving` and have been deprecated for at least 1 minor releases.
+
+The legacy configuration file `flink-conf.yaml` is no longer supported. Please use `config.yaml` with standard YAML format instead. A migration tool is provided to convert a legacy `flink-conf.yaml` into a new `config.yaml`. See [Migrate from flink-conf.yaml to config.yaml](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#migrate-from-flink-confyaml-to-configyaml) for more details.
+
+Dedicated APIs for configuring specific options are removed from `StreamExecutionEnvironment` and `ExecutionConfig`. All options should now be set via `Configuration` and `ConfigOption`. See also the list of [removed programming APIs](#list-of-removed-programming-apis).
```

Comment:
> All options should now be set via `Configuration` and `ConfigOption`

It's not accurate because Flink still offers convenience configuration methods like `setAutoWatermarkInterval()` in `ExecutionConfig`. The goal of the removal, however, is to prevent users from setting custom objects into Flink as configuration, which worse the configuration visibility.

Reply:
I see. How about:
```
Configuration APIs that takes Java objects as arguments are removed from `StreamExecutionEnvironment` and `ExecutionConfig`. They should now be set via `Configuration` and `ConfigOption`. See also the list of [removed programming APIs](#list-of-removed-programming-apis).
```

Reply:
LGTM.